### DISCUSSION TITLE
How do I move one commit from a branch to its parent?

### DISCUSSION THREAD

**Discussion by Artamus** [VirtusLab/git-machete#1252] [Q&A] [answered]
Given the following structure:
```
  main
  |
  | oof1
  o-split-1 (untracked)
    |
    | oof2
    | oof22
    o-split-2 * (untracked)
```
I would like to move the commit `oof2` to `split-1`. Now, I know I could set the fork point of `split-2` to `oof2`, which would make it not-part of `split-2`, but I cannot figure out which command is supposed to add it to `split-1`. 
The way I figured I could do this right now is to checkout the specific commit that should get its own branch, create a new branch from it , add it to git machete, play with fork points, use `advance` and play with fork points some more.
I have to be missing an easier way to do this 😅.

**Comment by PawelLipski** [comment, marked as answer]
👋🏻 IIUC, I think the easiest approach would be something like:
```shell
git checkout split-1
git merge --ff-only split-2~
# and, to rectify fork point of split-1
git machete fork-point --override-to-parent
```

Let know if it achieves your purpose!

**Reply by Artamus** [reply]
This works perfectly, thank you so much, Paweł!

I have to say, of the tools I've evaluated so far doing "stacked diffs" workflows, git-machete has been the best. I'm personally more partial to having 1 commit equal 1 review, but the general UX of git machete makes me a lot more comfortable trying to sell this workflow to my team.
Thanks for the great work!

**Reply by PawelLipski** [reply]
Aww thanks 🥰 I'll appreciate if you spread the word in your team!
